src/GuiBase/Viewer/Gizmo/Gizmo.cpp

```diff
@@ -68,20 +89,22 @@ bool Gizmo::findPointOnPlane( const Engine::Camera& cam,
     return hasHit;
 }
 
-void Gizmo::addRenderObject( Engine::RenderObject* ro, const std::shared_ptr<Engine::Mesh>& mesh ) {
-    m_meshes.push_back( mesh );
-    m_renderObjects.push_back( m_comp->addRenderObject( ro ) );
+void Gizmo::addRenderObject( Engine::RenderObject* ro ) {
+    m_comp->addRenderObject( ro );
+    m_ros.push_back( ro );
 }
 
-Engine::RenderTechnique* Gizmo::makeRenderTechnique( const std::string& mtlName,
-                                                     bool rtPerVertexColor ) {
+void Gizmo::changeMat( uint ro, uint mat ) {
+    m_ros[ro]->setMaterial( s_material[mat] );
```

Comment:
I wonder if it's better to change the associated material, or just change the color of the material.
If it work like this, I'm ok to accept the PR, but without performance test please add a todo to check which is better in futur investigation.

Reply:
I'm against this proposal. It's better to only change the color of the material rather than the whole material (and then the rendertechique). See my overall comment.